src/Locator.php

```diff
@@ -71,14 +75,20 @@ class Locator implements RegistryAware
     public $dom;
     protected $registry;
 
-    public function __construct(\SimplePie\File $file, $timeout = 10, $useragent = null, $max_checked_feeds = 10, $force_fsockopen = false, $curl_options = [])
+    /**
+     * @var Client
+     */
+    private $http_client = null;
+
+    public function __construct(\SimplePie\File $file, $timeout = 10, $useragent = null, $max_checked_feeds = 10, $force_fsockopen = false, $curl_options = [], ?Client $http_client = null)
```

Comment:
Ugly but I do not see a better way to make it preserve compatibility. Other than maybe using `func_get_args` and introspection. But that would be even uglier.

Reply:
Another idea that I had would be to allow the storage of instances in the `Registry`. This way we could get the http client from the registry instead of setting it in the constructor. But that could be done in another PR.